tree/dataframe/test/dataframe_utils.cxx

```diff
@@ -233,3 +235,21 @@ TEST(RDataFrameUtils, TypeName2TypeID)
    // TODO(jblomer): Ideally, we would want the next one not to throw an exception
    EXPECT_THROW(RDFInt::TypeName2TypeID("std::vector<std::vector<float>>"), std::runtime_error);
 }
+
+TEST(RDataFrameUtils, JittedFilters)
+{
+   auto df0 = ROOT::RDataFrame(10).Define("x", [] { return 42; });
+   df0.Snapshot("t", "f.root");
+   auto df1 = ROOT::RDataFrame(10).Define("x", [] { return -42; });
+   df1.Snapshot("fr", "fr.root");
+
+   TFile f("f.root");
+   TTree *t = (TTree *)f.Get<TTree>("t");
+   t->AddFriend("fr", "fr.root");
+   auto df2 = ROOT::RDataFrame(*t);
```

Comment:
we don't need to write out files for this (which makes the test much much slower):

```suggestion
   TTree t("t", "t");
   int x = 42;
   t.Branch("x", &x);
   t.Fill();
   TTree fr("fr", "fr");
   x = -42;
   fr.Branch("x", &x);
   fr.Fill();
   t.AddFriend(fr);

   ROOT::RDataFrame df(*t);
```